lib/features/impact_groups/cubit/impact_groups_cubit.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app/core/enums/country.dart';
+import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app/features/impact_groups/models/impact_group.dart';
+import 'package:givt_app/features/impact_groups/repo/impact_groups_repository.dart';
+
+part 'impact_groups_state.dart';
+
+class ImpactGroupsCubit extends Cubit<ImpactGroupsState> {
+  ImpactGroupsCubit(
+    this._impactGroupInviteRepository,
+    this._authCubit,
+  ) : super(const ImpactGroupsState()) {
+    _authCubit.stream.listen((event) async {
+      if (event.status == AuthStatus.authenticated &&
+          event.user.country == Country.us.countryCode) {
+        await fetchImpactGroups();
+      }
+    });
+  }
+  final ImpactGroupsRepository _impactGroupInviteRepository;
+  final AuthCubit _authCubit;
+  Future<void> fetchImpactGroups() async {
+    emit(state.copyWith(status: ImpactGroupCubitStatus.loading));
+
+    try {
+      final impactGroups =
+          await _impactGroupInviteRepository.fetchImpactGroups();
+
+      emit(
+        state.copyWith(
+          status: ImpactGroupCubitStatus.fetched,
+          impactGroups: impactGroups,
+        ),
+      );
+    } catch (e) {
+      emit(
+        state.copyWith(
+          status: ImpactGroupCubitStatus.error,
+          error: e.toString(),
+        ),
+      );
+    }
+  }
+
+  void checkForInvites() {
+    for (var i = 0; i < state.impactGroups.length; i++) {
+      if (state.impactGroups[i].status == ImpactGroupStatus.invited) {
+        emit(
```

Comment:
just wondering how the logic works for multiple invitation at the same time?

Reply:
Only one bottomsheet will pop up, the next one will pop up the next time they go to home page or they logout/login